src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -274,55 +338,16 @@ impl A320Hydraulic {
             lgciu2,
         );
 
-        if number_of_steps_floating_point < 1.0 {
-            // Can't do a full time step
-            // we can decide either do an update with smaller step or wait next iteration
-            // for now we only update lag accumulator and chose a hard fixed step: if smaller
-            // than chosen time step we do nothing and wait next iteration
-
-            // Time lag is float part only of num of steps (because is < 1.0 here) * fixed time step to get a result in time
-            self.lag_time_accumulator = Duration::from_secs_f64(
-                number_of_steps_floating_point * min_hyd_loop_timestep.as_secs_f64(),
-            );
-        } else {
-            // Int part is the actual number of loops to do
-            // rest of floating part goes into accumulator
-            let num_of_update_loops = number_of_steps_floating_point.floor() as u32;
-
-            self.lag_time_accumulator = Duration::from_secs_f64(
-                (number_of_steps_floating_point - (num_of_update_loops as f64))
-                    * min_hyd_loop_timestep.as_secs_f64(),
-            ); // Keep track of time left after all fixed loop are done
-
-            // Then run fixed update loop for main hydraulics
-            for _ in 0..num_of_update_loops {
-                // First update what is currently consumed and given back by each actuator
-                // Todo: might have to split the actuator volumes by expected number of loops
-                self.update_actuators_volume();
-
-                self.update_fixed_step(
-                    &context.with_delta(min_hyd_loop_timestep),
-                    engine1,
-                    engine2,
-                    overhead_panel,
-                    engine_fire_push_buttons,
-                    lgciu1,
-                    lgciu2,
-                );
-            }
-
-            // This is the "fast" update loop refreshing ACTUATORS_SIM_TIME_STEP_MULT times faster
-            // here put everything that needs higher simulation rates like physics solving
-            let num_of_actuators_update_loops =
-                num_of_update_loops * Self::ACTUATORS_SIM_TIME_STEP_MULTIPLIER;
-
-            // If X times faster we divide step by X
-            let delta_time_physics =
-                min_hyd_loop_timestep / Self::ACTUATORS_SIM_TIME_STEP_MULTIPLIER;
-            for _ in 0..num_of_actuators_update_loops {
-                self.update_fast_rate(&context, &delta_time_physics);
-            }
-        }
+        // Here we update at a fixed rate only. Mainly core hydraulics
+        self.update_fixed_step(
+            &context,
+            engine1,
+            engine2,
+            overhead_panel,
+            engine_fire_push_buttons,
+            lgciu1,
+            lgciu2,
+        );
```

Comment:
**suggestion**: Consider changing the structure of the `update` function, such that the amount of "steps" taken is only known at this level and doesn't leak further down. For example:

```rust
while let Some(step_delta) = self.physics_updater.next() {
    self.update_physics(&context.with_delta(step_delta))
}

while let Some(step_delta) = self.some_new_updater_which_does_it_per_frame.next() {
    self.update_something(
            &context.with_delta(step_delta),
            &overhead_panel,
            &autobrake_panel,
            rat_and_emer_gen_man_on,
            emergency_elec_state,
            lgciu1,
            lgciu2,
    );
}

while let Some(step_delta) = self.core_hydraulic_updater.next() {
    self.update_core_hydraulics(
            &context.with_delta(step_delta)
            engine1,
            engine2,
            overhead_panel,
            engine_fire_push_buttons,
            lgciu1,
            lgciu2,
    );
}
```

Note that this example hasn't yet taken into account the `Iterator` comment I made elsewhere.

Reply:
Thanks, was obvious to do it this way, I totally brain farted!

Reply:
@crocket63 Unresolved this one for now. The implementation still leaks, in the sense that the function names at the lower level are still called `update_max_fixed_step`, `update_every_frame`, and `update_fixed_step`, thereby removing the point of the exercise of moving the "execution control" to a higher level.